### ISSUE TITLE
ChiSquaredTest#goodness_of_fit gives ZeroDivisionError

### ISSUE THREAD

**Issue by nfproc** [estebanz01/ruby-statistics#78]
ChiSquaredTest#goodness_of_fit gives an error with some distributions. The shortest code I could reproduce the error is as follows:
```
irb(main):001:0> require 'statistics'
=> true
irb(main):002:0> StatisticalTest::ChiSquaredTest.goodness_of_fit(0.01, 477, [481, 483, 482, 488, 478, 471, 477, 479, 475, 462])
        9: from /usr/bin/irb:23:in `<main>'
        8: from /usr/bin/irb:23:in `load'
        7: from /usr/lib/ruby/gems/2.7.0/gems/irb-1.2.1/exe/irb:11:in `<top (required)>'
        6: from (irb):2
        5: from /var/lib/gems/2.7.0/gems/ruby-statistics-3.0.0/lib/statistics/statistical_test/chi_squared_test.rb:28:in `goodness_of_fit'
        4: from /var/lib/gems/2.7.0/gems/ruby-statistics-3.0.0/lib/statistics/distribution/chi_squared.rb:14:in `cumulative_function'
        3: from /var/lib/gems/2.7.0/gems/ruby-statistics-3.0.0/lib/math.rb:49:in `lower_incomplete_gamma_function'
        2: from /var/lib/gems/2.7.0/gems/ruby-statistics-3.0.0/lib/math.rb:27:in `simpson_rule'
        1: from /var/lib/gems/2.7.0/gems/ruby-statistics-3.0.0/lib/math.rb:27:in `/'
ZeroDivisionError (divided by 0)
```
Perhaps a problem lies in lib/math.rb:49. It looks to cause a zero division if x < 0.5.

Thank you in advance.

**Comment by estebanz01** [comment]
Thanks for your report. I'll look at it!

**Comment by estebanz01** [comment]
OK, the chi score for this input is `52/53` or `0.9811320754716981` with 9 degrees of freedom. It's failing when trying to calculate the probability. I'll keep digging.

**Comment by nfproc** [comment]
Thank you for information, which made the problem more clear.

It looks that the chi score divided by 2 (in this case `0.490566...`) is given as x of  the 
`lower_incomplete_gamma_function`. That function calculates  `(10_000 * x.round).round` that becomes 0 in this case. Probably this is why the zero division occurred at lib/math.rb:27.

**Comment by estebanz01** [comment]
@nfproc is it possible for you to test my fix in #79 ? if you are using a `Gemfile`, you just need to `gem 'ruby-statistics', github: 'estebanz01/ruby-statistics', branch: 'issue-78'`.

Let me know if it's possible, else I'd run some basic tests then release a new version.

Thanks again for reporting this !

**Comment by nfproc** [comment]
I confirmed that the problem has been solved with both the piece of code (presented above) and my application.
Thank you for quick fix!
